### PR TITLE
Deduct gas fee from zkauth address

### PR DESCRIPTION
**Pull request by Mdaiki0730**
## Description

Deduct the gas fee that would have been deducted by DeductFeeDecorator from zkauth address
Mostly the same as DeductFeeDecorator.AnteHandle.
https://github.com/Finschia/finschia-sdk/blob/df06ee16ed85f660b391f69d0732d6b00560c5fa/x/auth/ante/fee.go#L75

In the future there should be a way to branch out with zkauth more generally.

## Checklist:

- [ ] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/zkauth/ante/zkauth.go [hunk 1]
```diff
@@ -152,7 +154,63 @@ func (zdf ZKAuthDeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
...
 
-	// Case of zkauth msg, does nothing in this case
+	// Almost no different from the implementation of zdf.dfd.AnteHandle, uses ZKauthAddress for feePayer.
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "Tx must be a FeeTx")
+	}
+
+	if addr := zdf.ak.GetModuleAddress(authtypes.FeeCollectorName); addr.Empty() {
+		return ctx, fmt.Errorf("fee collector module account (%s) has not been set", authtypes.FeeCollectorName)
```
Comment (shiki-tak) [inline x/zkauth/ante/zkauth.go:164]: How about using sdkerrors.Wrap like elsewhere?

File: x/zkauth/ante/zkauth.go [hunk 2]
```diff
@@ -152,7 +154,63 @@ func (zdf ZKAuthDeductFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
...
+	}
+
+	fee := feeTx.GetFee()
+	signature := zkMsgs[0].GetZkAuthSignature()
```
Comment (shiki-tak) [inline x/zkauth/ante/zkauth.go:168]: I'm not sure, so I have a question, is it not necessary to run a loop with zkMsgs?Is zkMsgs[0] enough?

**Review by shiki-tak — COMMENTED** [review]
Please add CHANGELOG

File: x/zkauth/ante/zkauth.go (hunk 2 above)
Comment (Mdaiki0730) [reply x/zkauth/ante/zkauth.go:168]: The honest answer is that the specifications have not been determined at the moment.
I would like to note in the comments that it is necessary to consider how to respond when multiple zkmsg are included in one tx.

File: x/zkauth/ante/zkauth.go (hunk 1 above)
Comment (Mdaiki0730) [reply x/zkauth/ante/zkauth.go:164]: This is not an error case for user input.
I can't find a suitable sdkerror, so how about leave it to a higher-level function?

Comment (shiki-tak) [reply x/zkauth/ante/zkauth.go:164]: Surely, or If the result of this function is Empty(), other modules seem to panic.

**Review by shiki-tak — DISMISSED** [review]

File: x/zkauth/ante/zkauth.go (hunk 2 above)
Comment (zemyblue) [reply x/zkauth/ante/zkauth.go:169]: And we need to check the case that the tx messagea are mixed both zkauth and other normal messages.

**Review by zemyblue — APPROVED** [review]
But I agree this pr is good. but we need to check the case that the messages are added multiple and mixing message type. So please add another issue for solving multiple messages.

**Comment by Mdaiki0730** [comment]
> So please add another issue for solving multiple messages.

I added an issue, so I'll merge this PR. Thank you.
https://github.com/Finschia/finschia-sdk/issues/1286

**Review by zemyblue — APPROVED** [review]

**Pull request merged**